lambda/translquote.ml

```diff
@@ -2839,15 +2843,23 @@ and quote_expression_desc transl stage e =
         let names_defs =
           List.map
             (fun vb ->
+              let cstr =
+                match vb.vb_pat.pat_extra with
+                | [(Tpat_constraint ct, _, _)] -> Some ct
+                | [] -> None
+                | _ -> assert false
```

Comment:
While there are a few `assert false`s in Translquote, I'd prefer we didn't and that they all were fatal errors that tell you where the fatal error happened e.g. "Translquote: let_" and print the location (a concurrent PR is doing this)